internal/wasm/host.go

```diff
@@ -0,0 +1,390 @@
+package internalwasm
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"reflect"
+
+	"github.com/tetratelabs/wazero/wasm"
+)
+
+type HostFunction struct {
+	Name         string
+	FunctionType *FunctionType
+	GoFunc       *reflect.Value
```

Comment:
I guess these can be unexported?

Reply:
I can take a look. In general post this change I think a lot of code can be refactored, but export matters less since internal.

Reply:
indeed! unexpected

Reply:
unexported I meant